stable/pgadmin/templates/deployment.yaml

```diff
@@ -51,24 +61,32 @@ spec:
             - name: pgadmin-data
               mountPath: /var/lib/pgadmin
           resources:
-            {{- toYaml .Values.resources | nindent 12 }}
+            {{- .Values.resources | toYaml | nindent 12 }}
       volumes:
         - name: pgadmin-data
-        {{- if .Values.persistence.enabled }}
+        {{- if .Values.persistentVolume.enabled }}
           persistentVolumeClaim:
-            claimName: {{ $fullName }}
+            claimName: {{ if .Values.persistentVolume.existingClaim }}{{ .Values.persistentVolume.existingClaim }}{{- else }}{{ $fullName }}{{- end }}
         {{- else }}
           emptyDir: {}
         {{- end }}
-      {{- with .Values.nodeSelector }}
+    {{- if .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- .Values.iamgePullSecrets | toYaml | nindent 8 }}
+    {{- end }}
+    {{- if .Values.nodeSelector }}
       nodeSelector:
-        {{- toYaml . | nindent 8 }}
-      {{- end }}
-    {{- with .Values.affinity }}
+        {{- .Values.nodeSelector | toYaml | nindent 8 }}
+    {{- end }}
+    {{- if .Values.securityContext }}
+      securityContext:
+        {{- .Values.securityContext | toYaml | nindent 8 }}
+    {{- end }}
+    {{- if .Values.affinity }}
       affinity:
-        {{- toYaml . | nindent 8 }}
+        {{- .Values.affinity | toYaml | nindent 8 }}
     {{- end }}
-    {{- with .Values.tolerations }}
```

Comment:
more of a personal taste - but I favour using `with` as it is easier to read and keep consistent due to no duplication.

Reply:
Would you like me to adjust this? Personally I found this statement more clear. But I don't mind adjusting it back.